src/Tests/PubkeyEncryptTestBase.php

```diff
@@ -52,4 +53,25 @@ protected function initializePubkeyEncrypt() {
     }
   }
 
+  /**
+   * Custom login function to ensure the handling of cookies after a user login.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   User object representing the user to log in.
+   */
+  protected function drupalLogin(AccountInterface $account) {
+    parent::drupalLogin($account);
+
+    // When a user logs in, his Private key gets temporarily stored in a cookie
+    // and should be present there till he logs out. Since SimpleTest by default
+    // does not provide the functionality of retaining cookies during curl
+    // requests, hence manually doing it here as it is necessary.
+    foreach ($this->cookies as $name => $value) {
+      // We're only concerned with the cookie containing Private key for a user.
+      if (preg_match('/.private_key/', $name, $matches)) {
+        $_COOKIE[$name] = urldecode($value['value']);
```

Comment:
WebTestBase::$cookies are urlencoded and hence using the urldecode function here. 

@nerdstein, @colans do you think I should create an issue in the Drupal core to get WebTestBase::$cookies urldecoded by default? This would make it easy for people to use WebTestBase::$cookies without manually applying the urldecode function on them. But I'm not sure whether this issue is worth pursuing.